azurerm/internal/services/datafactory/data_factory_dataset_binary_resource.go

```diff
@@ -0,0 +1,400 @@
+		Type:          &linkedServiceType,
+	}
+
+	description := d.Get("description").(string)
+	binaryTableset := datafactory.BinaryDataset{
+		BinaryDatasetTypeProperties: &binaryDatasetProperties,
+		LinkedServiceName:           linkedService,
+		Description:                 &description,
+	}
+
+	if v, ok := d.GetOk("folder"); ok {
+		name := v.(string)
+		binaryTableset.Folder = &datafactory.DatasetFolder{
+			Name: &name,
+		}
+	}
+
+	if v, ok := d.GetOk("parameters"); ok {
+		binaryTableset.Parameters = expandDataFactoryParameters(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("annotations"); ok {
+		annotations := v.([]interface{})
+		binaryTableset.Annotations = &annotations
+	}
+
+	if v, ok := d.GetOk("additional_properties"); ok {
+		binaryTableset.AdditionalProperties = v.(map[string]interface{})
+	}
+
+	datasetType := string(datafactory.TypeBasicDatasetTypeBinary)
+	dataset := datafactory.DatasetResource{
+		Properties: &binaryTableset,
+		Type:       &datasetType,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.FactoryName, id.Name, dataset, ""); err != nil {
+		return fmt.Errorf("creating/updating Data Factory Dataset Binary  %q (Data Factory %q / Resource Group %q): %s", id.Name, id.FactoryName, id.ResourceGroup, err)
+	}
+
+	_, err := client.Get(ctx, id.ResourceGroup, id.FactoryName, id.Name, "")
+	if err != nil {
+		return fmt.Errorf("retrieving Data Factory Dataset Binary %q (Data Factory %q / Resource Group %q): %s", id.Name, id.FactoryName, id.ResourceGroup, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceDataFactoryDatasetBinaryRead(d, meta)
+}
+
+func resourceDataFactoryDatasetBinaryRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.DatasetClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataSetID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.FactoryName, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Data Factory Dataset Binary %q (Data Factory %q / Resource Group %q): %s", id.Name, id.FactoryName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("data_factory_name", id.FactoryName)
+
+	binaryTable, ok := resp.Properties.AsBinaryDataset()
+	if !ok {
+		return fmt.Errorf("classifiying Data Factory Dataset Binary %q (Data Factory %q / Resource Group %q): Expected: %q Received: %q", id.Name, id.FactoryName, id.ResourceGroup, datafactory.TypeBasicDatasetTypeBinary, *resp.Type)
+	}
+
+	d.Set("additional_properties", binaryTable.AdditionalProperties)
+
+	if binaryTable.Description != nil {
+		d.Set("description", binaryTable.Description)
+	}
+
+	parameters := flattenDataFactoryParameters(binaryTable.Parameters)
+	if err := d.Set("parameters", parameters); err != nil {
```

Comment:
not sure we need to assign these to local variables first?
```suggestion
	if err := d.Set("parameters", flattenDataFactoryParameters(binaryTable.Parameters)); err != nil {
```

Reply:
No, ty!